src/main/java/TaskList.java

```diff
@@ -0,0 +1,179 @@
+                    return date;
+                } else {
+                    toAdd = new Event(ls[0], date);
+                    storage.add(toAdd);
+                }
+
+            } else {
+                throw new DukeException("Sorry, please specify event date after \"/at\".");
+            }
+            break;
+        default:
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :(");
+        }
+
+        return ui.addTaskLine(toAdd, storage.size());
+    }
+
+    /**
+     * Completes given from the task list.
+     * @param s Index of task to be completed on the list
+     * @throws DukeException The Exception of Duke bot
+     */
+    public String doneTask(String s) throws DukeException {
+        try {
+            int i = Integer.parseInt(s);
+            if (i < 1 || i > storage.size()) {
+                throw new DukeException("You have entered an invalid number: " + i
+                    + ". Please try again.");
+            } else {
+                Task t = storage.get(i - 1);
+                Task completed = t.setDone(true);
+                storage.set(i - 1, completed);
+                return "Nice! I've marked this task as done:\n" + "  "
+                        + completed + "\n";
+            }
+        } catch (NumberFormatException nfe) {
+            return "Please state the completed task number after \"done\".\n";
+        }
+    }
+
+    /**
+     * Deletes given task from the task list.
+     * @param s Index of task to be completed on the list
+     * @throws DukeException The Exception of Duke bot
+     */
+    public String delTask(String s) throws DukeException {
+        if (storage.size() == 0 || s.toLowerCase().equals("all")) {
+            storage.clear();
+            return "All tasks cleared!!\n";
+        } else if (s.equals("")) {
+            try {
+                return addTask("delete", "");
+            } catch (DukeException e) {
+                return e.getMessage();
+            }
+        } else {
+            try {
+                int i = Integer.parseInt(s);
+                if (i < 1 || i > storage.size()) {
+                    throw new DukeException("You have entered an invalid number: " + i
+                        + ". Please try again.");
+                } else {
+                    Task t = storage.get(i - 1);
+                    storage.remove(i - 1);
+                    return ui.removeTaskLine(t, storage.size());
+                }
+            } catch (NumberFormatException e) {
+                return "Please state the completed task number after \"delete\".\n";
+            }
+        }
+    }
+
+    /**
+     * Prints the list of tasks that contain the given string.
+     * @param s The string that should appear in the task
+     */
+    public String findTask(String s) {
+        if (storage.isEmpty()) {
+            return "Your list is empty!\n";
+        } else {
+            int index = 1;
```

Comment:
What do you think of using the variable name `count` here to differentiate between the index of the task list and index of the matching tasks?